history-sync.plugin.zsh

```diff
@@ -178,32 +172,35 @@ history_sync_pull() {
     [[ -o extendedhistory ]] && _squash_multiline_commands_in_files
 
     # Merge
-    CAT "$ZSH_HISTORY_FILE" "$ZSH_HISTORY_FILE_DECRYPT_NAME" | \
-      AWK '/:[0-9]/ { if(s) { print s } s=$0 } !/:[0-9]/ { s=s"\n"$0 } END { print s }' | \
-      SORT -u > "$ZSH_HISTORY_FILE_MERGED_NAME"
-    MV "$ZSH_HISTORY_FILE_MERGED_NAME" "$ZSH_HISTORY_FILE"
-    RM  "$ZSH_HISTORY_FILE_DECRYPT_NAME"
+    cat "$ZSH_HISTORY_FILE" "$ZSH_HISTORY_FILE_DECRYPT_NAME" | \
+      AWK '/:[0-9]/ { if(s) { print s } s=$0 } !/:[0-9]/ { s=s"\n"$0 } END { print s }' \
+      | SORT -u > "$ZSH_HISTORY_FILE_MERGED_NAME"
+    mv "$ZSH_HISTORY_FILE_MERGED_NAME" "$ZSH_HISTORY_FILE"
+    rm  "$ZSH_HISTORY_FILE_DECRYPT_NAME"
     cd  "$DIR"
 
     # Check if EXTENDED_HISTORY is enabled, and if so, restore multi-line
     # commands in the local history file
     [[ -o extendedhistory ]] && _restore_multiline_commands_in_file
+    # Remove trailing empty new lines from multi-line commands
+    SED -E -i ':a;N;$!ba;s/(\\+\n)+\n/\n/g' "$ZSH_HISTORY_FILE"
```

Comment:
This will remove line end backslashes everywhere...

```shell
$ cat tmp.txt
: 1766012538:0;cat ~/.zsh_history
: 1766012585:0;echo hi \\

: 1766012590:0;cat ~/.zsh_history
: 1766013445:0;echo hi \\
more!
: 1766013451:0;cat ~/.zsh_history
$ sed -E ':a;N;$!ba;s/(\\+\n)|\n/\n/g' tmp.txt
: 1766012538:0;cat ~/.zsh_history
: 1766012585:0;echo hi

: 1766012590:0;cat ~/.zsh_history
: 1766013445:0;echo hi
more!
: 1766013451:0;cat ~/.zsh_history
```

So, it has to be more selective.